File: 110-balanced-binary-tree/balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        def solve(root):
            if not root:
                return 0
            lh=solve(root.left)
            if lh==-1:
                return -1
            rh=solve(root.right)
            if rh==-1:
                return -1
            if abs(lh-rh)>1:
                return -1
            else :
                return max(rh,lh)+1
        k=solve(root)
        if k ==-1 :
            return False 
        else :
            return True
            

File: 110-balanced-binary-tree/balanced-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
int ng(TreeNode* root)
{
    if (root==nullptr)
    return 0;
    if (ng(root->left)==-1)
    return -1;
    if (ng(root->right)==-1)
    return -1;
    if (abs(ng(root->right)-ng(root->left))>1)
    return -1;
    return 1+max(ng(root->right),ng(root->left));


}
    bool isBalanced(TreeNode* root) {
        int a=ng(root);
        if (a==-1)
        return false;
        return true;
    }
};